lib/AmdMainTemplatePlugin.js

```diff
@@ -16,7 +16,7 @@ class AmdMainTemplatePlugin {
 	apply(compilation) {
 		const mainTemplate = compilation.mainTemplate;
 
-		compilation.templatesPlugin("render-with-entry", (source, chunk, hash) => {
+		compilation.tapMainAndChunkTemplates("AmdMainTemplatePlugin", "renderWithEntry", (source, chunk, hash) => {
```

Comment:
Use
``` js
for(const template of [compilation.mainTemplate, compilation.chunkTemplate]) {
  template.hooks.renderWithEntry(...);
}
```

Reason: I want the `.hooks.xxx` visible in source code to be able to search for `.hooks`.

Reply:
Alternative:

We add a `MultiHook` which works like this:

``` js
// Compilation contructor
this.bothTemplates = {
  hooks: {
    renderWithEntry: new MultiHook([
      compilation.mainTemplate.renderWithEntry, 
      compilation.chunkTemplate.renderWithEntry
    ])
  }
}
```

Reply:
I think I like your first suggestion because it allows us to keep the hooks in one place.